dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/JdbcEnrollmentAnalyticsManager.java

```diff
@@ -342,6 +357,119 @@ protected String getSelectClause( EventQueryParams params )
         return "select " + StringUtils.join( selectCols, "," ) + " ";
     }
 
+    /**
+     * Returns an encoded column name respecting the geometry/coordinate format.
+     * The given QueryItem must be of type COORDINATE.
+     *
+     * @param item the {@link QueryItem}
+     * @return the column selector or EMPTY if the item valueType is not
+     *         COORDINATE. ie.: ( select '[' ||
+     *         round(ST_X("GyJHQUWZ9Rl")::numeric, 6) || ',' ||
+     *         round(ST_Y("GyJHQUWZ9Rl")::numeric, 6) || ']' as "GyJHQUWZ9Rl"
+     *         from analytics_event_qDkgAbB5Jlk where
+     *         analytics_event_qDkgAbB5Jlk.pi = ax.pi and "SzVk2KvkSSd" is not
+     *         null and ps = 'hYyB7FUS5eR' order by executiondate desc limit 1 )
+     *
+     * @throws NullPointerException if item is null
+     */
+    @Override
+    protected String getCoordinateColumn( final QueryItem item )
+    {
+        return getCoordinateColumn( item, null );
+    }
+
+    /**
+     * Returns an encoded column name respecting the geometry/coordinate format.
+     * The given QueryItem can be of type COORDINATE or ORGANISATION_UNIT.
+     *
+     * @param suffix is currently ignored. Not currently used for enrollments
+     * @param item the {@link QueryItem}
+     * @return the column selector or EMPTY if the item valueType is not
+     *         COORDINATE. ie.: ( select '[' || round(ST_X("colName")::numeric,
+     *         6) || ',' || round(ST_Y("colName")::numeric, 6) || ']' as
+     *         "colName" from analytics_event_qDkgAbB5Jlk where
+     *         analytics_event_qDkgAbB5Jlk.pi = ax.pi and "colName" is not null
+     *         and ps = 'hYyB7FUS5eR' order by executiondate desc limit 1 )
+     *
+     * @throws NullPointerException if item is null
+     */
+    @Override
+    protected String getCoordinateColumn( final QueryItem item, final String suffix )
+    {
+        if ( item.getProgram() != null )
+        {
+            final String eventTableName = ANALYTICS_EVENT + item.getProgram().getUid();
+            final String colName = quote( item.getItemId() );
+
+            String psCondition = "";
+
+            if ( item.hasProgramStage() )
+            {
+                assertProgram( item );
+
+                psCondition = "and ps = '" + item.getProgramStage().getUid() + "' ";
+            }
+
+            String stCentroidFunction = "";
+
+            if ( ValueType.ORGANISATION_UNIT == item.getValueType() )
+            {
+                stCentroidFunction = "ST_Centroid";
+
+            }
+
+            return "(select " +
```

Comment:
Is it possible to format this in a more readable way? Or is the auto-formatter deciding it?

Reply:
Done